src/main/ruby/core/integer.rb

```diff
@@ -180,6 +180,12 @@ def pred
     self - 1
   end
 
+  def digits base = 10
+    raise Math::DomainError, 'out of domain' if negative?
+
+    to_s(base.to_int).chars.map(&:to_i).reverse
```

Comment:
```ruby
base = Truffle::Type.coerce_to_int(base)
```
is slightly better than just calling `to_int`, as it checks the result is a Fixnum as expected and throws TypeError if coercion doesn't work.